[Mohammed Faiz] iP

Mohammed-Faizzzz opened the pull request:
# Duke Bartholomew Hamish Montgomery (AKA Bart)

Bart frees your mind of having to remember things you need to do. It's:

* text-based
* easy to learn
* _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Mohammed-Faizzzz committed:
Completed Level-0

Mohammed-Faizzzz committed:
Level 1 completed

Mohammed-Faizzzz committed:
Completed Level-2

Mohammed-Faizzzz committed:
Completed Level-3

Mohammed-Faizzzz committed:
Completed Level-4

Mohammed-Faizzzz committed:
Completed A-TestUiTesting

Mohammed-Faizzzz committed:
code cleanup; OOP

Mohammed-Faizzzz committed:
Completed Level 5

Mohammed-Faizzzz committed:
completed Level-6

Mohammed-Faizzzz committed:
added delete functionality to Level-6

Mohammed-Faizzzz committed:
Add File object

Mohammed-Faizzzz committed:
Completed Level-7

Mohammed-Faizzzz committed:
Use deleteOnExit method

Mohammed-Faizzzz committed:
Add Dates and Times

Mohammed-Faizzzz committed:
Create OOP Classes

Mohammed-Faizzzz committed:
Add Ui Class functionality

Mohammed-Faizzzz committed:
Add TaskList Class functionality

Mohammed-Faizzzz committed:
Add Parser Class Functionality

Mohammed-Faizzzz committed:
Add task handlers

Mohammed-Faizzzz committed:
Fix Parser bugs

Mohammed-Faizzzz committed:
Fix TaskList bugs

Mohammed-Faizzzz committed:
Create package duke

Mohammed-Faizzzz committed:
Open Gradle Support

Mohammed-Faizzzz committed:
Merge branch 'master' into add-gradle-support

Update add-gradle-support branch

Mohammed-Faizzzz committed:
Modify Gradle Functionality

Mohammed-Faizzzz committed:
Add TaskListTest.java

Mohammed-Faizzzz committed:
Add ParserTest

Mohammed-Faizzzz committed:
Add JUnit Test in TaskListTest

Mohammed-Faizzzz committed:
Implement Find

Mohammed-Faizzzz committed:
Add Javadocs for Duke

Mohammed-Faizzzz committed:
Add Javadocs for Deadline

Mohammed-Faizzzz committed:
Add Javadocs for Events

Mohammed-Faizzzz committed:
Add Javadocs for Parser

Mohammed-Faizzzz committed:
Add Javadocs for TaskList

Mohammed-Faizzzz committed:
Add Javadocs for Task

Mohammed-Faizzzz committed:
Add Javadocs for Storage

Mohammed-Faizzzz committed:
Add Javadocs for ToDos

Mohammed-Faizzzz committed:
Add Javadocs for ToDos

Mohammed-Faizzzz committed:
Merge A-JavaDoc and Master

Mohammed-Faizzzz committed:
Apply Coding Standard for Duke

Mohammed-Faizzzz committed:
Apply Coding Standards

Mohammed-Faizzzz committed:
Merge A-CodingStandard and Master

Mohammed-Faizzzz committed:
Add JUnit Tests for DeadlineTest

Mohammed-Faizzzz committed:
Add JUnit Tests for EventTest

Mohammed-Faizzzz committed:
Add JUnit Tests for ToDoTest

Mohammed-Faizzzz committed:
Add JUnit Tests for UiTest

Mohammed-Faizzzz committed:
Fix find function bug

Mohammed-Faizzzz committed:
Add CheckStyle

Mohammed-Faizzzz committed:
Merge branch-A-CheckStyle

Mohammed-Faizzzz committed:
Add basic GUI features

Mohammed-Faizzzz committed:
Add images and response

Mohammed-Faizzzz committed:
Format GUI for Duke

wujy28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+ * The Task class represents a generic task item with a description and completion status.
+ */
+public class Task {
+    Boolean done = false;
```
Might be better if the boolean name sounds like a boolean (according to the Textbook), something like `isDone` :)

wujy28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+ */
+public class Task {
+    Boolean done = false;
+    String task;
```
I think the textbook recommends having a name that is a bit more descriptive and can explain the variable to the reader more accurately. Maybe something like `taskDescription` might be more suitable?

wujy28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Marks the task as incomplete.
+     */
+    public void unMark() {
```
This is just a nitpick but I think 'unmark' can be considered a word and so maybe the method name could have been just `unmark()` HAHA

wujy28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+     *
+     * @return The horizontal line as a string.
+     */
+    public String line() {
```
The textbook recommends using verbs for method names, so it might be better to name this method something like `getLine()`?

wujy28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+     *
+     * @return The formatted string representation of the task.
+     */
+    public String getTask() {
```
It might just be me but I feel like the method name is a bit misleading 😅, as it sounds like it will return a task object and not a string. Perhaps a name like `toFormattedString()` would be better (or anything else you can think of haha)?

wujy28 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,249 @@
+ * The TaskList class manages a list of tasks and provides methods to interact with and manipulate the tasks.
+ */
+public class TaskList {
+    static File taskList;
```
I think this variable name is also a little misleading as it sounds like it's a TaskList object and not a File 😅

wujy28 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,249 @@
+        taskList = new File("./src/main/data/tasklist.txt");
+        taskList.deleteOnExit();
+    }
+    static ArrayList<Task> tasks = new ArrayList<>();
```
I'm not sure whether this is part of the coding standard or coding quality guidelines but it might be neater if this variable is lumped together with the other class variables at the top :)

wujy28 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,34 @@
+     * Displays a greeting message to the user.
+     */
+    public void greet() {
+        String greeting = line + "I extend to you my utmost felicitations, User! I am " + name + "." + "\n" + "What may I do for you?" + "\n" + line;
```
I think the line is a little too long here (>120 characters)? You can use line wrapping :)

wujy28 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+     *
+     * @return True if the user input is "bye", indicating an exit command; false otherwise.
+     */
+    public boolean bye() {
```
I feel like the method names here are a little misleading as well; even though the names are verbs but they sound more like they will perform the action of listing, marking etc. instead of returning a boolean. Maybe something like `isBye()` will be more suitable?

wujy28 reviewed (commented):
Generally well done! :)
However please pay attention to certain naming conventions and follow the guidelines provided by the Textbook (under 'Code Quality')

Mohammed-Faizzzz committed:
Add GUI

Mohammed-Faizzzz committed:
Add Main, MainWindow and fxml files

Mohammed-Faizzzz committed:
Integrate Duke to GUI

Mohammed-Faizzzz committed:
Add OOP: make Boolean field done private.

Previously, Boolean field done was package private and classes Deadline, Event, and Todo modified the field directly.
Making the field private and using the superclass method to modify it adheres to OOP principles

Mohammed-Faizzzz committed:
Delete dead code:

Previous iterations of Duke used the Duke class as the main class, and used the startChat function that did not require an input.
With the new iteration making use of the GUI, the previous code becomes redundant. Deleting this dead code improves code quality.

Mohammed-Faizzzz committed:
Improve Code Quality and Style for Ui Class

Mohammed-Faizzzz committed:
Improve Code Quality and Style for Storage Class

Mohammed-Faizzzz committed:
Complete A-CodeStyle

Mohammed-Faizzzz committed:
Add assertions to TaskList

Mohammed-Faizzzz committed:
Merge pull request #1 from Mohammed-Faizzzz/branch-A-Assertion

Marge branch-A-Assertion to Master branch

Mohammed-Faizzzz committed:
Merge pull request #2 from Mohammed-Faizzzz/branch-A-CodeStyle

Merge branch-A-CodeStyle with master branch

Mohammed-Faizzzz committed:
Create Contact Class

Mohammed-Faizzzz committed:
Add getters to Contact

Mohammed-Faizzzz committed:
Complete Extension C-Sort

Mohammed-Faizzzz committed:
Make GUI Modifications

Mohammed-Faizzzz committed:
Update README.md

Mohammed-Faizzzz committed:
Update README.md

Complete Features Portion

Mohammed-Faizzzz committed:
Update README.md

Add User Guide for todo

Mohammed-Faizzzz committed:
Update README.md

Complete User Guide

Mohammed-Faizzzz committed:
Complete UserGuide

Mohammed-Faizzzz committed:
Improve GUI, OOP of Ui Class

Mohammed-Faizzzz committed:
Switch to branch-A-EvenBetterGui

Mohammed-Faizzzz committed:
Improve GUI

Mohammed-Faizzzz committed:
Add OOP to Task and child classes

Mohammed-Faizzzz committed:
Add Javadocs to Dialogbox Class

Mohammed-Faizzzz committed:
Add Javadocs to DukeException Class

Mohammed-Faizzzz committed:
Add Javadocs for Task Class

Mohammed-Faizzzz committed:
Merge branch 'branch-A-EvenBetterGui'

Merge to update master with GUI Changes and OOP Changes

Mohammed-Faizzzz committed:
Add Ui.png

Mohammed-Faizzzz committed:
Delete unnecessary code

Mohammed-Faizzzz committed:
Add Error Handling

Mohammed-Faizzzz committed:
Improve sorting function

Mohammed-Faizzzz committed:
Move Parsing methods to Parser

Mohammed-Faizzzz committed:
Add Javadocs

Mohammed-Faizzzz committed:
Add OOP for TaskList and Ui

Mohammed-Faizzzz committed:
add Javadocs

Mohammed-Faizzzz committed:
Resolve merge conflicts

Mohammed-Faizzzz committed:
Bug fixes

Mohammed-Faizzzz committed:
Add JUnit Tests

Mohammed-Faizzzz committed:
Complete JUnit Test

Mohammed-Faizzzz committed:
Add tasklist.txt

Mohammed-Faizzzz committed:
Fix Bugs

Mohammed-Faizzzz committed:
Bug Fixes

Mohammed-Faizzzz committed:
Modify ReadME

Mohammed-Faizzzz committed:
Bug Fixes

Mohammed-Faizzzz committed:
Update README.md

Mohammed-Faizzzz committed:
Update ReadME

Mohammed-Faizzzz committed:
Update README.md

Mohammed-Faizzzz committed:
Update ReadME.md

Mohammed-Faizzzz committed:
Add tasklist.txt

Mohammed-Faizzzz committed:
Modify tasklist.txt

Mohammed-Faizzzz committed:
Add More OOP

Mohammed-Faizzzz committed:
Resolve hardcoded file path

Mohammed-Faizzzz committed:
Resolve Empty List Error Message

Mohammed-Faizzzz committed:
Merge branch-A-StorageHandling to master

Mohammed-Faizzzz committed:
Javadocs modification

Mohammed-Faizzzz committed:
Change dependencies to acceptable version

Mohammed-Faizzzz committed:
Add gradle.yml

Mohammed-Faizzzz committed:
Edit gradle.yml

Mohammed-Faizzzz committed:
Change StorageTest Class

Mohammed-Faizzzz committed:
Edit StorageTest

Mohammed-Faizzzz committed:
Edit StorageTest

Mohammed-Faizzzz committed:
Modify StorageTest

Mohammed-Faizzzz committed:
Edit build.gradle
